src/main/java/com/monumental/controllers/UserController.java

```diff
@@ -0,0 +1,41 @@
+package com.monumental.controllers;
+
+import com.monumental.controllers.helpers.CreateUserRequest;
+import com.monumental.exceptions.InvalidEmailOrPasswordException;
+import com.monumental.exceptions.UnauthorizedException;
+import com.monumental.models.User;
+import com.monumental.repositories.UserRepository;
+import com.monumental.services.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class UserController {
+
+    @Autowired
+    UserService userService;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @PostMapping("/api/signup")
+    public User signup(@RequestBody CreateUserRequest user) throws InvalidEmailOrPasswordException {
+        return this.userService.signup(user);
+    }
+
+    @GetMapping("/api/session")
+    @PreAuthorize("hasRole('COLLABORATOR')")
+    public User getSession() throws UnauthorizedException {
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (!(principal instanceof UserDetails)) {
+            throw new UnauthorizedException();
+        }
```

Comment:
What do these lines do?

Reply:
So basically, you *should* only be able to access this method if you're logged in because of the `@PreAuthorize` above, but sometimes this method was getting executed when logged out (probably just while I was testing things, don't think it does anymore) and in those cases `principal` is the string `"anonymousUser"`